lib/events.js

```diff
@@ -20,6 +20,17 @@ EventEmitter.prototype._maxListeners = undefined;
 // added to it. This is a useful default which helps finding memory leaks.
 EventEmitter.defaultMaxListeners = 10;
 
+function ensureEvents(emitter) {
+  if (!emitter._events ||
+      emitter._events === Object.getPrototypeOf(emitter)._events) {
```

Comment:
This won't work for code like below, I think?  Instances of B won't have an `._events` property on their immediate prototype.

```javascript
function A() { EventEmitter.call(this); }
A.prototype = Object.create(EventEmitter.prototype);
A.prototype._events = {};

function B() { A.call(this); }
B.prototype = Object.create(A.prototype);
```